src/FakeItEasy.Analyzer.Shared/ArgumentConstraintTypeMismatchCodeFixProvider.cs

```diff
@@ -111,38 +111,34 @@ private static async Task<Document> MakeNotNullConstraintAsync(CodeFixContext co
             var constraintNode = GetConstraintNode(diagnostic, root);
 
             // The A<T> type
-            var aType = (constraintNode as MemberAccessExpressionSyntax)?.Expression as GenericNameSyntax;
+            var aType = (GenericNameSyntax)((MemberAccessExpressionSyntax)constraintNode).Expression;

```

Comment:
Maybe this is a bad change. Is there a case where a code fix will be provided source that doesn't match the expected form?
That is, will we ever _not_ get a `GenericNameSyntax`?